### PR TITLE
Automatically detect if Spot has the arm installed.

### PR DESCRIPTION
**Pull request by gremigi-bdai**
With this PR, we do not need the argument `has_arm` anymore. When we execute `bdai spot driver <spot-name>` this parameter is retrieved form Spot itself using the `SpotWrapper` interface.

### CODE REVIEW

File: spot_driver/launch/spot_driver.launch.py [hunk 1]
```diff
@@ -133,15 +131,25 @@ def create_rviz_config(robot_name: str) -> None:
 
 
 def launch_setup(context: LaunchContext, ld: LaunchDescription) -> None:
+    logger = logging.getLogger("spot_driver_launch")
+
```
Comment (amessing-bdai) [inline spot_driver/launch/spot_driver.launch.py:135]: I think this was unintentionally left here

**Review by amessing-bdai — COMMENTED** [review]
Overall looks good. A logger that isn't used was left.

File: spot_driver/launch/spot_driver.launch.py (hunk 1 above)
Comment (gremigi-bdai) [reply spot_driver/launch/spot_driver.launch.py:135]: It is required few lines below by the `SpotWrapper`. I tried passing None but it fails at runtime.

**Review by jbarry-bdai — COMMENTED** [review]
Yay!  Tests to do:

- [ ] Spot without arm launches
- [x] Spot without arm is shown without arm in rviz
- [x] Spot with arm launches
- [x] Spot with arm is shown with arm in rviz
- [x] For spot with arm all camera topics are published for the hand camera including depth registered (and we actually see images on these topics)

**Comment by gremigi-bdai** [comment]
Spot with arm launches correctly and the arm is displayed in RVIZ.

image

**Review by bhung-bdai — APPROVED** [review]
LGTM. Reminder to make the changes to the cli args to deprecate the `has-arm` field when we run within the docker and to update the `gitman.yml` file in the main repo as well.

**Pull request merged**